hack/coverage.sh

```diff
@@ -10,7 +10,7 @@ GINKGO_COVERAGE_ARGS="-cover -coverprofile=${COVERAGE_FILE} -outputdir=. --skipP
 GINKGO_ARGS="-v -r --progress ${GINKGO_EXTRA_ARGS} ${GINKGO_COVERAGE_ARGS}"
 
 # source files excluded from ginkgo coverage report. these files are not used during the unit test and include code that is only relevant to the installed product.
-declare -a EXCLUDE_FILES_FROM_COVERAGE=("nodemaintenance_controller_init.go")
+declare -a EXCLUDE_FILES_FROM_COVERAGE=("nodemaintenance_controller_init.go" "zz_generated.deepcopy.go" "nodemaintenance_webhooks.go")
```

Comment:
nodemaintenance_webhooks.go is not run as part of the unit test. it contains stuff needed by the sigs controller-runtime for doing the http server for serving webhook requests from apiserver.

Reply:
> nodemaintenance_webhooks.go is not run as part of the unit test.

yes, but why?

> it contains stuff needed by the sigs controller-runtime for setting up the http server for serving webhook requests (these requests originate from the apiserver)

I get that. But how is that a reason for being excluded from unit tests?

> Including this file would just skew the coverage statistics of the unit tests

Sorry, but that isn't a very good argument for excluding files from unit tests / coverage reports.

This is code written by us. It should be unit tested. In case there is a good reason to not do so, we have an explanation for coverage getting lower. But excluding files from coverage looks like faking our own coverage stats so they just look good but don't have a good value anymore.